core/modules/installer/installer.browser.inc

```diff
@@ -0,0 +1,470 @@
+ */
+function installer_browser_get_queued_releases() {
+  $releases = array();
+
+  if (isset($_SESSION['installer_browser_install_releases_list'])) {
+    foreach ($_SESSION['installer_browser_install_releases_list'] as $release_name => $project) {
+      if (is_array($project) AND !empty($project)) {
+        $releases[$release_name] = $project;
+      }
+    }
+  }
+
+  return $releases;
+}
+
+/**
+ * Fetches results from the server based on the parameters passed in.
+ *
+ * $filters should be an associative array with the following keys:
+ * array(
+ *   'version' => '1', // The Major Version of Backdrop that is running on the Client
+ *   'text' => 'views', // The text that was entered as the search query, or '' if none
+ *   'type' => 'module', // The type of project being searched
+ *   'page' => 3, // The zero-based page number
+ *   'items_per_page' => 12, // How many results are requested per page
+ * )
+ *
+ * The installer_browser_fetch_results($filters) call returns an array like this:
+ *
+ * array(
+ *   'total' = 5, // The total number of results found for the filters
+ *   'projects' => array( // An array of projects returned for this page request
+ *     'views' => array( // A project array keyed by the machine name
+ *       'type' => 'module', // The type of project this is. Can be 'module' or 'theme'
+ *       'title' => 'Views', // The title of the project
+ *       'name' => 'views', // The machine name of the project
+ *       'author' => 'merlinofchaos', // The author's name
+ *       'description' => "Long project description ...",
+ *       'image' => 'http://www.example.com/image.jpg', // Absolute url to the image, if any
+ *       'usage' => '542312', // How many Downloads the module has had
+ *       'project url' => 'http://www.<MAINSITE_URL>/projects/views', // Absolute url to the project page, if any
+ *       'project status url' => 'http://updates.<MAINSITE_URL>/release-history/views/1.x', // The absolute url of the update checker, formatted like how <MAINSITE_URL> Update Status does it
+ *       'last updated' => '12342523', // UNIX Timestamp of when the project was last updated
+ *       'maintenance status' => 'Actively maintained', // Maintenance status
+ *       'development status' => 'Under active development', // Development status
+ *       'rating' => '9.6', // A rating on a scale of 1 to 10 of the project, if available
+ *       'dependencies' => array( // An array of the dependencies of this module
+ *         'entity',
+ *       ),
+ *     ),
+ *   'another_project => array( ... ),
+ *   ),
+ * );
+ *
+ * @param array $filters
+ *   An associative array of queries to use to filter results.
+ *
+ * @return
+ *   Returns an array of results.
+ */
+function installer_browser_fetch_results($filters) {
+  $server = installer_browser_get_server();
+  // Attempt to retrieve the cached version of this page.
+  $cid = md5(serialize(array_merge($filters, $server)));
+  if ($cache = cache_get($cid)) {
```

Comment:
These cache IDs are weird. A dedicated cache bin might not be necessary, but they should at least be prefixed with something that identifies them as coming from Installer, e.g. `$cid = 'installer:results:md5(serialize(array_merge($filters, $server)));`. That makes it possible to clear browser results only.